pygit2/index.py

```diff
@@ -286,6 +286,27 @@ def diff_to_tree(self, tree, flags=0, context_lines=3, interhunk_lines=0):
 
         return Diff.from_c(bytes(ffi.buffer(cdiff)[:]), self._repo)
 
+    def diff_index_entries(self, ancestor, ours, theirs):
+        """diff_index_entries(ancestor, ours, theirs) -> str
+
+        Diff index entries.
+
+        Return a :py:class:`str` object with the differences between
+        the given index entries.
+        """
+        cmergeresult = ffi.new('git_merge_file_result *')
+
+        cancestor = ancestor._to_c()[0] if ancestor is not None else ffi.NULL
+        cours = ours._to_c()[0] if ours is not None else ffi.NULL
+        ctheirs = theirs._to_c()[0] if theirs is not None else ffi.NULL
+
+        err = C.git_merge_file_from_index(
+                cmergeresult, self._repo._repo,
+                cancestor, cours, ctheirs,
+                ffi.NULL);
+        check_error(err)
+
+        return ffi.string(cmergeresult.ptr)
```

Comment:
Fixed now. This also revealed some kind of bug where the ffi.string would get scrambled at the end when not creating it with cmergeresult.len. Seems as if the string is not correctly NULL-terminated in xdiff. I didn't dive into the issue, though.